File: Spoj/GCD2_GCD2.py
import sys
# sys.stdin = open("/home/shiva/Learning/1.txt", "r")
# sys.stdout = open("/home/shiva/Learning/2.txt", "w")
def gcd(a,b):
	if b==0:
		return a
	return gcd(b,a%b)

t = int(input())
for i in range(t):
	a, b = map(int,input().split())
	print(gcd(a,b))

File: Spoj/GCD2_GCD2.cpp
#include <bits/stdc++.h>
#define vi vector<int>
#define pi pair<int,int>
#define vii vector<pi>
#define ll long long
using namespace std;
int main() {
#ifndef ONLINE_JUDGE
	freopen("/home/shiva/Learning/1.txt", "r", stdin);
	freopen("/home/shiva/Learning/2.txt", "w", stdout);
#endif
	ios::sync_with_stdio(false); cin.tie(0);
	int t; cin >> t;
	while (t--) {
		int a; string b; cin >> a >> b;
		if (a == 0) {cout << b << endl; continue;}
		int ib = 0;
		for (auto i : b)
			ib = (ib * 10 + (i - '0')) % a;
		while (ib) {
			int temp = a % ib;
			a = ib;
			ib = temp;
		}
		cout << a << endl;
	}

}